### PR TITLE
CNF-13067: O-RAN V3 Rest Api: Status Notification

### PR DESCRIPTION
**Pull request by jzding**
- Remove local store for pubsub: sub.json. Now access to sub info only through SubscriberAPI/store.
- Update UriLocation to use service name.
- Remove subject from Cloud Event
- Remove datacontenttype from Cloud Event
- Reconstruct configMap for subscription info
- Fix response code in subscription failure cases.
- Added unit tests

More details in doc https://docs.google.com/document/d/1aIgGWhFDDz9aI7mIZm_nlkb99UrV7hv5gS5m_XShCt4/edit

### CODE REVIEW

File: v2/routes.go [hunk 1]
```diff
@@ -67,26 +67,19 @@ func (s *Server) createSubscription(w http.ResponseWriter, r *http.Request) {
 		localmetrics.UpdateSubscriptionCount(localmetrics.FAILCREATE, 1)
 		return
 	}
-	if subExists, ok := s.pubSubAPI.HasSubscription(sub.GetResource()); ok {
+	clientIDs := s.subscriberAPI.GetClientIDByResource(sub.GetResource())
```
Comment (aneeshkp) [inline v2/routes.go:70]: why Plural clientID's?

Comment (aneeshkp) [reply v2/routes.go:70]: should be one clientID for the resource

File: vendor/github.com/redhat-cne/sdk-go/pkg/subscriber/subscriber_reader.go [hunk 2]
```diff
@@ -31,9 +31,8 @@ func (s *Subscriber) GetSubStore() *store.PubSubStore {
 
 // CreateCloudEvents ...
 func (s *Subscriber) CreateCloudEvents() (*cloudevents.Event, error) {
-	ce := cloudevents.NewEvent(cloudevents.VersionV03)
-	ce.SetDataContentType(cloudevents.ApplicationJSON)
-	ce.SetSpecVersion(cloudevents.VersionV03)
+	ce := cloudevents.NewEvent(cloudevents.VersionV1)
+	ce.SetSpecVersion(cloudevents.VersionV1)
```
Comment (aneeshkp) [inline vendor/github.com/redhat-cne/sdk-go/pkg/subscriber/subscriber_reader.go:35]: why not setting content type?

File: v2/routes.go (hunk 1 above)
Comment (jzding) [reply v2/routes.go:71]: This returns `clientIDs []uuid.UUID`. I did not change the logic.
https://github.com/redhat-cne/sdk-go/blob/main/v1/subscriber/subscriber.go#L254

File: vendor/github.com/redhat-cne/sdk-go/pkg/subscriber/subscriber_reader.go (hunk 2 above)
Comment (jzding) [reply vendor/github.com/redhat-cne/sdk-go/pkg/subscriber/subscriber_reader.go:35]: In O-RAN V3.0 spec, Event Data Model does not have `datacontenttype` and `subject`, so I removed them.

**Comment by aneeshkp** [comment]
Please add description to this PR

File: vendor/github.com/redhat-cne/sdk-go/pkg/subscriber/subscriber.go [hunk 3]
```diff
@@ -49,7 +49,7 @@ type Subscriber struct {
 	SubStore *store.PubSubStore `json:"subStore" omit:"empty"`
 	// EndPointURI - A URI describing the subscriber link .
 	// +required
-	EndPointURI *types.URI `json:"endPointURI" omit:"empty"`
```
Comment (aneeshkp) [inline vendor/github.com/redhat-cne/sdk-go/pkg/subscriber/subscriber.go:52]: is it possible the internal API direct call failure in older version ?

File: vendor/github.com/redhat-cne/sdk-go/pkg/subscriber/subscriber.go [hunk 4]
```diff
@@ -84,8 +84,8 @@ func (s *Subscriber) FailedCount() int {
 // String returns a pretty-printed representation of the Event.
 func (s *Subscriber) String() string {
 	b := strings.Builder{}
-	b.WriteString("  EndPointURI: " + s.GetEndPointURI() + "\n")
-	b.WriteString("  ID: " + s.GetClientID().String() + "\n")
```
Comment (aneeshkp) [inline vendor/github.com/redhat-cne/sdk-go/pkg/subscriber/subscriber.go:88]: Same here

File: vendor/github.com/redhat-cne/sdk-go/pkg/subscriber/subscriber_reader.go (hunk 2 above)
Comment (aneeshkp) [reply vendor/github.com/redhat-cne/sdk-go/pkg/subscriber/subscriber_reader.go:35]: Subject is used  by cloudEvent proxy  it is internally used 
https://github.com/redhat-cne/cloud-event-proxy/blob/d9e98a115ad764a2d15a4ab5b36594b2228d333c/examples/simplehttp/common/subscribe.go#L48

Comment (aneeshkp) [reply vendor/github.com/redhat-cne/sdk-go/pkg/subscriber/subscriber_reader.go:35]: also it is used by cloudevents specs https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md#example

Comment (jzding) [reply vendor/github.com/redhat-cne/sdk-go/pkg/subscriber/subscriber_reader.go:35]: removed ce.SetSubject() from simplehttp. I think we are fine here. The subject is not actually used in internal API. The ProcessOutChannel() only uses `Status` and `Action`.
https://github.com/redhat-cne/cloud-event-proxy/blob/main/cmd/main.go#L219

File: vendor/github.com/redhat-cne/sdk-go/pkg/subscriber/subscriber.go (hunk 3 above)
Comment (jzding) [reply vendor/github.com/redhat-cne/sdk-go/pkg/subscriber/subscriber.go:52]: no. tested with old version. The subscription was successful.

File: vendor/github.com/redhat-cne/sdk-go/pkg/subscriber/subscriber.go (hunk 4 above)
Comment (jzding) [reply vendor/github.com/redhat-cne/sdk-go/pkg/subscriber/subscriber.go:88]: no. tested with old version. The subscription was successful.

**Comment by jzding** [comment]
> Please add description to this PR

done

**Pull request merged**